src/table/components/TableWrapper.jsx

```diff
@@ -238,9 +238,15 @@ export default function TableWrapper(props) {
         data-testid="table-wrapper"
       >
         <Table stickyHeader aria-label={tableAriaLabel}>
-          <TableHeadWrapper {...props} setFocusedCellCoord={setFocusedCellCoord} focusedCellCoord={focusedCellCoord} />
+          <TableHeadWrapper
+            {...props}
+            paginationNeeded={paginationNeeded}
+            setFocusedCellCoord={setFocusedCellCoord}
+            focusedCellCoord={focusedCellCoord}
+          />
           <TableBodyWrapper
             {...props}
+            paginationNeeded={paginationNeeded}
```

Comment:
I don't think we should pass another prop here. I had a look at all the how we set this variable up top. We calculate it every time the TableWrapper re-renders. But it will only change if the data changes. So we might as well calculate it at that point and put it on tableData. Let me do a quick refactor and you can use that

Reply:
Sounds good